### PR TITLE
More basic tests

### PR DESCRIPTION
**Pull request by BCecatto**
###Why this PR?
Increase number of test

### CODE REVIEW

File: spec/refile/cloudinary_spec.rb [hunk 1]
```diff
@@ -1,7 +1,28 @@
...
+require "refile/cloudinary"
 
-describe Refile::Cloudinary do
+config = {
+  cloud_name: "hfkkd49r3",
+  api_key: "some-key",
+  api_secret: "some-secret"
+}
+
+RSpec.describe Refile::Cloudinary do
```
Comment (pusewicz) [inline spec/refile/cloudinary_spec.rb]: @BCecatto 👋 and thanks for the PR! Please move the `config` variable here as:

```ruby
let(:config) {{
  cloud_name: "hfkkd49r3",
  api_key: "some-key",
  api_secret: "some-secret"
}}
```

File: spec/refile/cloudinary_spec.rb [hunk 2]
```diff
@@ -1,7 +1,28 @@
...
   it "has a version number" do
     expect(Refile::Cloudinary::VERSION).not_to be nil
   end
+  
+  describe "behavior Refile::Cloudinary" do
+    let(:backend) { Refile::Cloudinary.new({**config}, max_size: 100 )}
+    
+    it "expect backend max size equal 100" do
+
```
Comment (pusewicz) [inline spec/refile/cloudinary_spec.rb]: ✂️ the empty line 😄

File: spec/refile/cloudinary_spec.rb [hunk 3]
```diff
@@ -1,7 +1,28 @@
...
+      expect(backend.max_size).to eq 100
+    end
+
+    it "object not nil" do
+
```
Comment (pusewicz) [inline spec/refile/cloudinary_spec.rb]: ✂️ the empty line here too! 😄

File: spec/refile/cloudinary_spec.rb [hunk 4]
```diff
@@ -1,7 +1,28 @@
...
+    let(:backend) { Refile::Cloudinary.new({**config}, max_size: 100 )}
+    
+    it "expect backend max size equal 100" do
+
+      expect(backend.max_size).to eq 100
+    end
+
+    it "object not nil" do
+      
+      expect(backend).to be_truthy
```
Comment (pusewicz) [inline spec/refile/cloudinary_spec.rb]: I'd probably change this to `.not_to be_nil` to match the test description.

**Review by pusewicz — CHANGES_REQUESTED** [review]
Thanks a lot for the PR! Please make these changes and I'll merge this 😸

File: spec/refile/cloudinary_spec.rb [hunk 5]
```diff
@@ -1,7 +1,34 @@
...
+  
+  describe "behavior Refile::Cloudinary" do
+    let(:backend) { Refile::Cloudinary.new( {**config} ) }
+
+    it "object not nil" do
+      expect(backend).not_to be_nil 
+    end
+
+    it "file max size 10 megabytes" do
+      expect(backend.max_size).to be <= 10_485_760
```
Comment (pusewicz) [inline spec/refile/cloudinary_spec.rb]: This test does not make much sense. I'd drop this one.

File: spec/refile/cloudinary_spec.rb [hunk 6]
```diff
@@ -1,7 +1,34 @@
...
+    end
+
+    it "file great than 10 megabytes is invalid" do
+      expect(backend.max_size).to_not be > 10_485_760
```
Comment (pusewicz) [inline spec/refile/cloudinary_spec.rb]: I'd drop this one too.

File: spec/refile/cloudinary_spec.rb [hunk 7]
```diff
@@ -1,7 +1,34 @@
...
+    end
+
+    it "default max_size 10 megabytes" do
+      expect(backend.max_size).to eq 10_485_760
+    end
```
Comment (pusewicz) [inline spec/refile/cloudinary_spec.rb]: This test is fine. I'd write another test where you initialise the backend with a different value for `max_size` and test that the accessor correctly returns it.

File: spec/refile/cloudinary_spec.rb [hunk 8]
```diff
@@ -1,7 +1,34 @@
...
+    api_secret: "some-secret"
+  } }
 
-describe Refile::Cloudinary do
   it "has a version number" do
     expect(Refile::Cloudinary::VERSION).not_to be nil
   end
+  
+  describe "behavior Refile::Cloudinary" do
+    let(:backend) { Refile::Cloudinary.new( {**config} ) }
```
Comment (pusewicz) [inline spec/refile/cloudinary_spec.rb]: Why `{**config}`? I'd simply pass `config` instead.

File: spec/refile/cloudinary_spec.rb [hunk 9]
```diff
@@ -1,7 +1,34 @@
...
+
+    it "object not nil" do
+      expect(backend).not_to be_nil
```
[the remark this answers is not in the document]
Comment (pusewicz) [reply spec/refile/cloudinary_spec.rb]: Please fix these automated messages.

File: refile-cloudinary.gemspec [hunk 10]
```diff
@@ -29,8 +29,8 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ["lib"]
 
-  spec.add_dependency             "cloudinary", "~> 1.1.7"
   spec.add_dependency             "refile", "~> 0.6.2"
+  spec.add_dependency             "cloudinary", "~> 1.1.7"
```
Comment (pusewicz) [inline refile-cloudinary.gemspec]: Why did you change the order? It was sorted alphabetically before.

**Review by pusewicz — CHANGES_REQUESTED** [review]
Thanks! I left some comments and I'd like those changes to be made before merging.

**Comment by BCecatto** [comment]
What need more?

File: spec/refile/cloudinary_spec.rb [hunk 11]
```diff
@@ -1,7 +1,27 @@
...
+    api_key: "some-key",
+    api_secret: "some-secret"
+  } }
 
-describe Refile::Cloudinary do
   it "has a version number" do
     expect(Refile::Cloudinary::VERSION).not_to be nil
   end
+  describe "behavior Refile::Cloudinary" do
+    let(:backend_default) { Refile::Cloudinary.new(**config) }
```
Comment (pusewicz) [inline spec/refile/cloudinary_spec.rb]: Please remove the `**`, they are not necessary and might be confusing to people looking at those.

File: spec/refile/cloudinary_spec.rb [hunk 12]
```diff
@@ -1,7 +1,27 @@
...
+    let(:backend_max_size_100) { Refile::Cloudinary.new({ **config }, max_size: 100) }
+    it "expect backend max size equal 10_485_760" do
+      expect(backend_default.max_size).to eq 10_485_760
+    end
+    it "object not nil" do
+      expect(backend_default).not_to be_nil 
+    end
+    it "change to 100 max_size" do
```
Comment (pusewicz) [inline spec/refile/cloudinary_spec.rb]: Please add new lines between the blocks.

File: spec/refile/cloudinary_spec.rb [hunk 13]
```diff
@@ -1,7 +1,27 @@
...
     expect(Refile::Cloudinary::VERSION).not_to be nil
   end
+  describe "behavior Refile::Cloudinary" do
+    let(:backend_default) { Refile::Cloudinary.new({ **config }) }
+    let(:backend_max_size_100) { Refile::Cloudinary.new({ **config }, max_size: 100) }
+    it "expect backend max size equal 10_485_760" do
+      expect(backend_default.max_size).to eq 10_485_760
+    end
+    it "object not nil" do
+      expect(backend_default).not_to be_nil
```
[the remark this answers is not in the document]
Comment (pusewicz) [reply spec/refile/cloudinary_spec.rb]: There's still some trailing space as you can see in the comment above.

**Comment by pusewicz** [comment]
Hey @BCecatto, thanks for the updates! I've added few new comments, we are really close to getting this merged!

File: spec/refile/cloudinary_spec.rb [hunk 14]
```diff
@@ -1,7 +1,30 @@
...
+  } }
 
-describe Refile::Cloudinary do
   it "has a version number" do
     expect(Refile::Cloudinary::VERSION).not_to be nil
   end
+  
+  describe "behavior Refile::Cloudinary" do
+    let(:backend_default) { Refile::Cloudinary.new(config) }
+    let(:backend_max_size_100) { Refile::Cloudinary.new({ **config }, max_size: 100) }
```
Comment (pusewicz) [inline spec/refile/cloudinary_spec.rb:17]: This like too, please :), no need for the `**`.

**Comment by sobrinho** [comment]
Can be closed in favor of #9.

**Comment by pusewicz** [comment]
Closing due to inactivity.

**Pull request closed without merging**